lib/hexpm_web/templates/dashboard/organization/index.html.eex

```diff
@@ -6,6 +6,92 @@
   </div>
 
   <div class="col-sm-9 organization members">
+    <div class="panel panel-default">
```

Comment:
Is it possible to extract this to a shared template between user and org. profiles? Or are there too many differences?

Reply:
I think it's better to have this duplication.
The main blockers for extracting a shared template are:
1. email fields for organization are `text_input`, but `select` for normal user
2. there is a `small` section saying `You can add or remove email address on the <a href="<%= Routes.email_path(Endpoint, :index) %>">email settings page</a>.` for normal user, but not for organization

We can add conditionals in the extracted shared template, but IMH it's better to leave them in separated templates.

Below is a diff between user profile form and organization profile form:

<details>

```diff
  ,*** /var/folders/m0/qgyn76495599mchnw44qlrnw0000gn/T/ediffXUV3nB	2020-01-10 20:30:41.000000000 +0800
  --- /var/folders/m0/qgyn76495599mchnw44qlrnw0000gn/T/ediffNBTt2D	2020-01-10 20:30:41.000000000 +0800
  ,***************
  ,*** 3,23 ****
          <div class="panel-body">
            <div class="form-group">
              <label>Profile picture</label>
  !           <img src="<%= gravatar_url(User.email(@current_user, :gravatar), :large) %>">
              <p>
                <small>
  !               <%= if User.email(@current_user, :gravatar) do %>
  !                 Gravatar is used to display your profile picture. You can choose your Gravatar email below or
  !                 go to <a href="https://en.gravatar.com/emails/">Gravatar</a> to change the image.
  !               <% else %>
  !                 <a href="https://en.gravatar.com/">Gravatar</a> is used to display your profile picture.
  !                 Choose a Gravatar email address below to show your avatar on your profile.
  !               <% end %>
                </small>
              </p>
            </div>

  !         <%= form_for @changeset, Routes.profile_path(Endpoint, :update), [method: :post], fn f -> %>
              <div class="form-group">
                <%= label f, :full_name %>
                <%= text_input f, :full_name %>
  --- 3,18 ----
          <div class="panel-body">
            <div class="form-group">
              <label>Profile picture</label>
  !           <img src="<%= gravatar_url(User.email(@organization.user, :gravatar), :large) %>">
              <p>
                <small>
  !               Gravatar is used to display organization profile picture. You can update the Gravatar email below or
  !               go to <a href="https://en.gravatar.com/emails/">Gravatar</a> to change the image.
                </small>
              </p>
            </div>

  !         <%= form_for @changeset, Routes.organization_path(Endpoint, :update_profile, @organization), [method: :post, as: :profile], fn f -> %>
              <div class="form-group">
                <%= label f, :full_name %>
                <%= text_input f, :full_name %>
  ,***************
  ,*** 26,41 ****

              <div class="form-group">
                <%= label f, :public_email %>
  !             <%= select f, :public_email, public_email_options(@changeset.data), value: public_email_value(@changeset.data) %>
                <%= error_tag f, :public_email %>
  -             <small>
  -               You can add or remove email address on the <a href="<%= Routes.email_path(Endpoint, :index) %>">email settings page</a>.
  -             </small>
              </div>

              <div class="form-group">
                <%= label f, :gravatar_email %>
  !             <%= select f, :gravatar_email, gravatar_email_options(@changeset.data), value: gravatar_email_value(@changeset.data) %>
                <%= error_tag f, :gravatar_email %>
              </div>

  --- 21,33 ----

              <div class="form-group">
                <%= label f, :public_email %>
  !             <%= text_input f, :public_email, placeholder: "Your organization's public email", value: @public_email %>
                <%= error_tag f, :public_email %>
              </div>

              <div class="form-group">
                <%= label f, :gravatar_email %>
  !             <%= text_input f, :gravatar_email, placeholder: "Your organization's gravatar email", value: @gravatar_email %>
                <%= error_tag f, :gravatar_email %>
              </div>

  ,***************
  ,*** 44,50 ****
                  <%= label f, :twitter %>
                  <div class="input-group">
                    <div class="input-group-addon">twitter.com/</div>
  !                 <%= text_input f, :twitter, placeholder: "your_twitter_id" %>
                  </div>

                  <%= error_tag f, :twitter %>
  --- 36,42 ----
                  <%= label f, :twitter %>
                  <div class="input-group">
                    <div class="input-group-addon">twitter.com/</div>
  !                 <%= text_input f, :twitter, placeholder: "organization_twitter_id" %>
                  </div>

                  <%= error_tag f, :twitter %>
  ,***************
  ,*** 54,60 ****
                  <%= label f, :github, "GitHub" %>
                  <div class="input-group">
                    <div class="input-group-addon">github.com/</div>
  !                 <%= text_input f, :github, placeholder: "your_github_id" %>
                  </div>
                  <%= error_tag f, :github %>
                </div>
  --- 46,52 ----
                  <%= label f, :github, "GitHub" %>
                  <div class="input-group">
                    <div class="input-group-addon">github.com/</div>
  !                 <%= text_input f, :github, placeholder: "organization_github_id" %>
                  </div>
                  <%= error_tag f, :github %>
                </div>
  ,***************
  ,*** 63,76 ****
                  <%= label f, :elixirforum, "Elixir Forum" %>
                  <div class="input-group">
                    <div class="input-group-addon">elixirforum.com/users/</div>
  !                 <%= text_input f, :elixirforum, placeholder: "your_elixir_forum_id" %>
                  </div>
                  <%= error_tag f, :elixirforum %>
                </div>

                <div class="form-group">
                  <%= label f, :freenode %>
  !               <%= text_input f, :freenode, placeholder: "Your nickname on Elixir IRC channel" %>
                  <span class="help-block">Elixir IRC channel:
                    <a href="irc://chat.freenode.net/elixir-lang">irc://chat.freenode.net/elixir-lang</a>
                  </span>
  --- 55,68 ----
                  <%= label f, :elixirforum, "Elixir Forum" %>
                  <div class="input-group">
                    <div class="input-group-addon">elixirforum.com/users/</div>
  !                 <%= text_input f, :elixirforum, placeholder: "organization_elixir_forum_id" %>
                  </div>
                  <%= error_tag f, :elixirforum %>
                </div>

                <div class="form-group">
                  <%= label f, :freenode %>
  !               <%= text_input f, :freenode, placeholder: "Your organization's nickname on Elixir IRC channel" %>
                  <span class="help-block">Elixir IRC channel:
                    <a href="irc://chat.freenode.net/elixir-lang">irc://chat.freenode.net/elixir-lang</a>
                  </span>
  ,***************
  ,*** 79,85 ****

                <div class="form-group">
                  <%= label f, :slack %>
  !               <%= text_input f, :slack, placeholder: "Your nickname on Elixir Slack channel" %>
                  <span class="help-block">Elixir Slack channel:
                    <a href="https://elixir-slackin.herokuapp.com">elixir-slackin.herokuapp.com</a>
                  </span>
  --- 71,77 ----

                <div class="form-group">
                  <%= label f, :slack %>
  !               <%= text_input f, :slack, placeholder: "Your organization's nickname on Elixir Slack channel" %>
                  <span class="help-block">Elixir Slack channel:
                    <a href="https://elixir-slackin.herokuapp.com">elixir-slackin.herokuapp.com</a>
                  </span>
```

</details>

Reply:
Thanks for checking.